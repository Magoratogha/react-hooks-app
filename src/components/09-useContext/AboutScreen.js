import { Fragment, useContext } from "react";
import { UserContext } from "./UserContext";

function AboutScreen() {

    const { user, setUser } = useContext(UserContext);

    return (
        <Fragment>
            <h1>AboutScreen</h1>
            <hr></hr>
            <pre>{ JSON.stringify(user, null, 3) }</pre>
            <button 
                onClick={ () => setUser({})}
                className="btn btn-outline-danger">
                    LogOut
            </button>
        </Fragment>
    );
}

export default AboutScreen;