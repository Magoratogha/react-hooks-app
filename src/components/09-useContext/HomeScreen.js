import { Fragment, useContext } from "react";
import { UserContext } from "./UserContext";

function HomeScreen() {

    const { user } = useContext(UserContext);

    return (
        <Fragment>
            <h1>HomeScreen</h1>
            <hr></hr>
            <pre>{ JSON.stringify(user, null, 3) }</pre>
        </Fragment>
    );
}

export default HomeScreen;