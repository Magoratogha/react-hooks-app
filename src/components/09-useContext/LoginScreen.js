import { Fragment, useContext } from "react";
import { UserContext } from "./UserContext";

function LoginScreen() {

    const { setUser } = useContext(UserContext);

    return (
        <Fragment>
            <h1>LoginScreen</h1>
            <hr></hr>
            <button 
                onClick={ () => setUser({ id: 1234, name: 'Magora' })}
                className="btn btn-outline-dark">
                    Login
            </button>
        </Fragment>
    );
}

export default LoginScreen;