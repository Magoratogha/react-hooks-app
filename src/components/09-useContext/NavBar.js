import { NavLink } from "react-router-dom";

function NavBar() {
    return (
        <nav className="navbar navbar-expand-sm navbar-dark bg-dark">
            <div className="container-fluid">
                <NavLink className="navbar-brand" to="/">useContext</NavLink>
                <div className="collapse navbar-collapse" id="navbarNavAltMarkup">
                    <div className="navbar-nav">
                        <NavLink to="/" className="nav-link">Home</NavLink>
                        <NavLink to="/about" className="nav-link">About</NavLink>
                        <NavLink to="/login" className="nav-link">Login</NavLink>
                    </div>
                </div>
            </div>
        </nav>
    );
}

export default NavBar;