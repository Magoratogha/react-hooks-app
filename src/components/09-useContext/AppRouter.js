import { BrowserRouter, Routes, Route } from "react-router-dom";
import AboutScreen from "./AboutScreen";
import LoginScreen from "./LoginScreen";
import HomeScreen from "./HomeScreen";
import NavBar from "./NavBar";

function AppRouter() {
    return(
        <BrowserRouter>
            <NavBar />
            <div className="m-4">
                <Routes>
                    <Route path="/" element={<HomeScreen />}></Route>
                    <Route path="/about" element={<AboutScreen />}></Route>
                    <Route path="/login" element={<LoginScreen />}></Route>
                    <Route path="*" element={<HomeScreen />}></Route>
                </Routes>
            </div>
        </BrowserRouter>
    );
}

export default AppRouter;