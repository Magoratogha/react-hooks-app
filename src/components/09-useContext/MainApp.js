import { useState } from "react";
import AppRouter from "./AppRouter";
import './styles.css';
import { UserContext } from "./UserContext";

function MainApp() {

    const [user, setUser] = useState({});

    return (
        <UserContext.Provider value={{ user, setUser }}>
            <AppRouter />
        </UserContext.Provider>

    );
}

export default MainApp;