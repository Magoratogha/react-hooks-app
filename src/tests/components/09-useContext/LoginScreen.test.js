import { render, screen } from "@testing-library/react";
import LoginScreen from "../../../components/09-useContext/LoginScreen";
import { UserContext } from "../../../components/09-useContext/UserContext";
import userEvent from "@testing-library/user-event";

describe('Test on <LoginScreen />', () => { 

    const setUser = jest.fn();
    const user = userEvent.setup();

    test('should render properly', () => {
        const { container } = render(
            <UserContext.Provider value={{ setUser }}>
                <LoginScreen />
            </UserContext.Provider>);
        expect(container).toMatchSnapshot();
    });

    test('should call setUser when button is clicked', async () => {
        const userMock = { id: 1234, name: 'Magora' };
        render(
            <UserContext.Provider value={{ setUser }}>
                <LoginScreen />
            </UserContext.Provider>);
        const loginButton = screen.queryByText('Login', { selector: 'button' });
        await user.click(loginButton);
        expect(setUser).toHaveBeenCalledWith(userMock);
    });
});