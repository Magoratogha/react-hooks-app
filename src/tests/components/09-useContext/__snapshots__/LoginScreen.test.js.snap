// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on <LoginScreen /> should render properly 1`] = `
<div>
  <h1>
    LoginScreen
  </h1>
  <hr />
  <button
    class="btn btn-outline-dark"
  >
    Login
  </button>
</div>
`;
